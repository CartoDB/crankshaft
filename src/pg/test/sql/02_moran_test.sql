\i test/fixtures/ppoints.sql
\i test/fixtures/ppoints2.sql

-- Areas of Interest functions perform some nondeterministic computations
-- (to estimate the significance); we will set the seeds for the RNGs
-- that affect those results to have repeateble results
SELECT cdb_crankshaft._cdb_random_seeds(1234);

SELECT ppoints.code, m.quads
  FROM ppoints
  JOIN cdb_crankshaft.CDB_AreasOfInterest_Local('SELECT * FROM ppoints', 'value') m
    ON ppoints.cartodb_id = m.ids
  ORDER BY ppoints.code;

SELECT cdb_crankshaft._cdb_random_seeds(1234);

SELECT ppoints2.code, m.quads
  FROM ppoints2
  JOIN cdb_crankshaft.CDB_AreasOfInterest_Local_Rate('SELECT * FROM ppoints2', 'numerator', 'denominator') m
    ON ppoints2.cartodb_id = m.ids
  ORDER BY ppoints2.code;
