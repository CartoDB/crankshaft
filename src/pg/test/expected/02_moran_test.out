\pset format unaligned
\set ECHO all
\i test/fixtures/ppoints.sql
SET client_min_messages TO WARNING;
\set ECHO none
_cdb_random_seeds

(1 row)
moran|significance
0.3399|-0.0196
(1 row)
_cdb_random_seeds

(1 row)
code|quads
01|HH
02|HL
03|LL
04|LL
05|LH
06|LL
07|HH
08|HH
09|HH
10|LL
11|LL
12|LL
13|HL
14|LL
15|LL
16|HH
17|HH
18|LL
19|HH
20|HH
21|LL
22|HH
23|LL
24|LL
25|HH
26|HH
27|LL
28|HH
29|LL
30|LL
31|HH
32|LL
33|HL
34|LH
35|LL
36|LL
37|HL
38|HL
39|HH
40|HH
41|HL
42|LH
43|LH
44|LL
45|LH
46|LL
47|LL
48|HH
49|LH
50|HH
51|LL
52|LL
(52 rows)
_cdb_random_seeds

(1 row)
code|quads
01|HH
02|HL
07|HH
08|HH
09|HH
13|HL
16|HH
17|HH
19|HH
20|HH
22|HH
25|HH
26|HH
28|HH
31|HH
33|HL
37|HL
38|HL
39|HH
40|HH
41|HL
48|HH
50|HH
(23 rows)
_cdb_random_seeds

(1 row)
code|quads
03|LL
04|LL
05|LH
06|LL
10|LL
11|LL
12|LL
14|LL
15|LL
18|LL
21|LL
23|LL
24|LL
27|LL
29|LL
30|LL
32|LL
34|LH
35|LL
36|LL
42|LH
43|LH
44|LL
45|LH
46|LL
47|LL
49|LH
51|LL
52|LL
(29 rows)
_cdb_random_seeds

(1 row)
code|quads
02|HL
05|LH
13|HL
33|HL
34|LH
37|HL
38|HL
41|HL
42|LH
43|LH
45|LH
49|LH
(12 rows)
_cdb_random_seeds

(1 row)
code|quads
01|HH
02|HL
03|LL
04|LL
05|LH
06|LL
07|HH
08|HH
09|HH
10|LL
11|LL
12|LL
13|HL
14|LL
15|LL
16|HH
17|HH
18|LL
19|HH
20|HH
21|LL
22|HH
23|LL
24|LL
25|HH
26|HH
27|LL
28|HH
29|LL
30|LL
31|HH
32|LL
33|HL
34|LH
35|LL
36|LL
37|HL
38|HL
39|HH
40|HH
41|HL
42|LH
43|LH
44|LL
45|LH
46|LL
47|LL
48|HH
49|LH
50|HH
51|LL
52|LL
(52 rows)
_cdb_random_seeds

(1 row)
code|quads
01|HH
02|HL
07|HH
08|HH
09|HH
13|HL
16|HH
17|HH
19|HH
20|HH
22|HH
25|HH
26|HH
28|HH
31|HH
33|HL
37|HL
38|HL
39|HH
40|HH
41|HL
48|HH
50|HH
(23 rows)
_cdb_random_seeds

(1 row)
code|quads
03|LL
04|LL
05|LH
06|LL
10|LL
11|LL
12|LL
14|LL
15|LL
18|LL
21|LL
23|LL
24|LL
27|LL
29|LL
30|LL
32|LL
34|LH
35|LL
36|LL
42|LH
43|LH
44|LL
45|LH
46|LL
47|LL
49|LH
51|LL
52|LL
(29 rows)
_cdb_random_seeds

(1 row)
code|quads
02|HL
05|LH
13|HL
33|HL
34|LH
37|HL
38|HL
41|HL
42|LH
43|LH
45|LH
49|LH
(12 rows)
